lib/gds_api/publishing_api_v2.rb

```diff
@@ -209,6 +209,50 @@ def get_links(content_id)
     get_json(links_url(content_id))
   end
 
+  # Update the links for a given content_id.
+  #
+  # Given a Content ID, it fetchs the existing link set and passes
+  # that to the provided block. The return value of the block is then
+  # passed to `patch_links`.
+  #
+  # @param content_id [String]
+  # @param options [Hash]
+  # @option params [Boolean] bulk_publishing Set to true to indicate
+  #   that this is part of a mass-republish. Allows the publishing-api
+  #   to prioritise human-initiated publishing (optional, default
+  #   false)
+  #
+  # @yieldparam [Hash] The existing links for the content_id.
+  # @yieldreturn [Hash] The new links for the content_id.
+  #
+  # @return [GdsApi::Response] The response from the patch links
+  # request.
+  #
+  # @example
+  #
+  #   publishing_api.update_links("a-content-id") do |links|
+  #     {
+  #       'link_type' => (
+  #         links.fetch('link_type', []) | ['e10655d3-31eb-4909-84f7-5b7a2312cbfc']
```

Comment:
I find this a little hard to follow because I always forget what `|` does. Maybe the specific use case from https://github.com/alphagov/content-tagger/blob/f43a884134e042afafca522c8c1631b7968ff7f2/app/services/tagging/tagger.rb#L4-L8 is a good example.

Reply:
Ok, I've changed the example to use `+` and `uniq`.